controllers/perconaservermysqlbackup_controller.go

```diff
@@ -315,3 +332,197 @@ func (r *PerconaServerMySQLBackupReconciler) getBackupSource(ctx context.Context
 
 	return src, nil
 }
+
+const (
+	finalizerDeleteBackup = "delete-backup"
+)
+
+func (r *PerconaServerMySQLBackupReconciler) checkFinalizers(ctx context.Context,
+	cr *apiv1alpha1.PerconaServerMySQLBackup) error {
+	if cr.DeletionTimestamp == nil {
+		return nil
+	}
+	l := log.FromContext(ctx).WithName("checkFinalizers")
+
+	finalizers := sets.NewString()
+	for _, finalizer := range cr.GetFinalizers() {
+		var err error
+		switch finalizer {
+		case finalizerDeleteBackup:
+			var ok bool
+			ok, err = r.deleteBackup(ctx, cr)
+			if !ok {
+				finalizers.Insert(finalizer)
+			}
+		default:
+			finalizers.Insert(finalizer)
+		}
+		if err != nil {
+			l.Error(err, "failed to run finalizer "+finalizer)
+			finalizers.Insert(finalizer)
+		}
+	}
+	cr.Finalizers = finalizers.List()
+
+	err := r.Update(ctx, cr)
+	if err != nil {
+		l.Error(err, "failed to update finalizers for backup", "backup", cr.Name)
+	}
+	return nil
+}
+
+func (r *PerconaServerMySQLBackupReconciler) sidecarBackupConfig(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQLBackup) (*apiv1alpha1.SidecarBackupConfig, error) {
+	conf := &apiv1alpha1.SidecarBackupConfig{
+		Destination: cr.Status.Destination,
+		VerifyTLS:   cr.Status.VerifyTLS,
+	}
+	nn := types.NamespacedName{
```

Comment:
It would be better to name it just 'n' or something else because generally in Go repeated letters like this represent a collection. Like `tt := []*Thing`
But it is minor, in this case maybe it is alright. Your call.

Reply:
I'll keep this naming because we use it across the code